### PR TITLE
VBM-89: add new item screen layout

### PR DESCRIPTION
**Pull request by natata7**
https://user-images.githubusercontent.com/10462332/186748351-0c78633d-e47a-4af7-9534-9243ce6af729.mov

### CODE REVIEW

File: packages/mobile/src/components/components.ts [hunk 1]
```diff
@@ -28,3 +29,4 @@ export { FlagBackgroundView } from './flag-background-view/flag-background-view'
 export { PrimaryButton } from './primary-button/primary-button';
 export { SecondaryButton } from './secondary-button/secondary-button';
 export { BuildInfo } from './build-info/build-info';
+export * from 'react-native-popover-view';
```
Comment (AleksandrTermenzhy) [inline packages/mobile/src/components/components.ts:32]:
```suggestion
export { default as Popover } from 'react-native-popover-view';
```

...and if you need types re-export them within `common/types`

File: packages/mobile/src/components/dropdown/dropdown.tsx [hunk 2]
```diff
@@ -34,16 +34,18 @@ const DropDown = <T extends FormControlValues>({
...
-        style={[
-          styles.label,
-          globalStyles.mb2,
-          globalStyles.fs12,
-          { color: colors.titlePrimary },
-        ]}
-      >
-        {label}
-      </Text>
+      {label && (
```
Comment (AleksandrTermenzhy) [inline packages/mobile/src/components/dropdown/dropdown.tsx:37]:
```suggestion
      {!!label && (
```

File: packages/mobile/src/components/icons/icons.tsx [hunk 3]
```diff
@@ -1,5 +1,6 @@
 import React, { FC } from 'react';
 import FeatherIcon from 'react-native-vector-icons/Feather';
+import Icon from 'react-native-vector-icons/FontAwesome5';
```
Comment (AleksandrTermenzhy) [inline packages/mobile/src/components/icons/icons.tsx:3]:
```suggestion
import FontAwesomeIcon from 'react-native-vector-icons/FontAwesome5';
```

File: packages/mobile/src/components/icons/icons.tsx [hunk 4]
```diff
@@ -66,6 +67,18 @@ const LogInIcon: AppIcon = (iconProps) => (
...
+  <FeatherIcon name="plus-circle" {...iconProps} />
+);
+
+const ImageIcon: AppIcon = (iconProps) => (
+  <FeatherIcon name="image" {...iconProps} />
+);
+
+const InfoIcon: AppIcon = (iconProps) => (
+  <Icon name="info-circle" {...iconProps} />
+);
```
Comment (AleksandrTermenzhy) [inline packages/mobile/src/components/icons/icons.tsx:80]: Let's use one icon set as long as it has required icons. I mean we may use a bit different-looking icon with the same meaning from the existing set

File: packages/mobile/src/components/input/input.tsx [hunk 5]
```diff
@@ -17,7 +17,7 @@ type Props<T extends FormControlValues> = {
   control: FormControl<T>;
   errors: FormControlErrors<T>;
   placeholder?: string;
-  contentContainerStyle?: ViewStyle;
+  contentContainerStyle?: ViewStyle | ViewStyle[];
```
Comment (AleksandrTermenzhy) [inline packages/mobile/src/components/input/input.tsx:20]:
```suggestion
  contentContainerStyle?: StyleProp<ViewStyle>;
```

File: packages/mobile/src/localization/languages/ua/screens.ts [hunk 6]
```diff
@@ -78,4 +78,31 @@ export const screens = {
...
+    TITLE_NAME_PLACEHOLDER: 'Введіть назву',
+    PRICE: 'Ціна',
+    PRICE_PLACEHOLDER: 'Введіть ціну',
+    PRICE_TOOLTIP: 'Інформація про ціну',
+    CURRENCY: 'Валюта',
+    CONTACT: 'Контактна інформація',
+    COUNTRY: 'Країна',
+    CITY: 'Місто',
+    MOBILE_PHONE: 'Телефон',
+    MOBILE_PHONE_PLACEHOLDER: '+380 ()',
```
Comment (AleksandrTermenzhy) [inline packages/mobile/src/localization/languages/ua/screens.ts:99]: This isn't seemed to require translation)

File: packages/mobile/src/localization/languages/ua/screens.ts [hunk 7]
```diff
@@ -78,4 +78,31 @@ export const screens = {
...
+    PHONE_TOOLTIP: 'Інформація про телефон',
+    CALLING_CODE: 'Код',
+    INSTAGRAM: 'Instagram',
+    INSTAGRAM_PLACEHOLDER: 'Ввідіть нікнейм',
```
Comment (AleksandrTermenzhy) [inline packages/mobile/src/localization/languages/ua/screens.ts:103]:
```suggestion
    INSTAGRAM_PLACEHOLDER: 'Введіть нікнейм',
```

File: packages/mobile/src/navigation/navigation.tsx [hunk 8]
```diff
@@ -54,6 +58,19 @@ const Navigation: FC = () => {
...
             component={SupportScreen}
           />
+          <Stack.Screen
+            name={RootScreenName.NEW_ITEM}
+            component={NewItemScreen}
+            options={{
+              title: t('make_a_post.TITLE'),
+              headerLeft: () => {
+                return <HeaderLeft />;
+              },
```
Comment (AleksandrTermenzhy) [inline packages/mobile/src/navigation/navigation.tsx:68]:
```suggestion
              headerLeft: HeaderLeft,
```

File: packages/mobile/src/screens/my-list/my-list.tsx [hunk 9]
```diff
@@ -1,10 +1,41 @@
 import React, { FC } from 'react';
-import { Text, ScreenWrapper } from '~/components/components';
+import { RootScreenName } from '~/common/enums/enums';
+import { RootNavigationProps } from '~/common/types/navigation/navigation-props';
```
Comment (AleksandrTermenzhy) [inline packages/mobile/src/screens/my-list/my-list.tsx:3]:
```suggestion
import { RootNavigationProps } from '~/common/types/types';
```

File: packages/mobile/src/screens/my-list/my-list.tsx [hunk 10]
```diff
@@ -1,10 +1,41 @@
...
+import { createStyles } from './styles';
 
 const MyList: FC = () => {
+  const navigation = useNavigation<RootNavigationProps>();
+  const { t } = useTranslation();
+  const { dark, colors } = useCustomTheme();
+  const user = useAppSelector(selectCurrentUser);
+
+  const styles = React.useMemo(() => createStyles(colors), [dark, colors]);
+  const handleNewItem = () => {
```
Comment (AleksandrTermenzhy) [inline packages/mobile/src/screens/my-list/my-list.tsx:26]:
```suggestion
  const handleCreateNewItemPress = () => {
```

File: packages/mobile/src/screens/new-item/components/new-item-form/new-item-form.tsx [hunk 11]
```diff
@@ -0,0 +1,207 @@
...
+  InfoIcon,
+  Text,
+  TouchableOpacity,
+  View,
+} from '~/components/components';
+import Popover from 'react-native-popover-view';
+import { useAppForm, useTranslation } from '~/hooks/hooks';
+import { CALLING_CODE, CITIES, COUNTRIES, CURRENCY } from '~/mock/new-item';
+import { globalStyles } from '~/styles/styles';
+import { ColorPalette } from '@vse-bude/shared';
```
Comment (AleksandrTermenzhy) [inline packages/mobile/src/screens/new-item/components/new-item-form/new-item-form.tsx:14]: Pls, respect the imports order

File: packages/mobile/src/screens/new-item/components/new-item-form/new-item-form.tsx [hunk 12]
```diff
@@ -0,0 +1,207 @@
...
+import { AddPhotos } from '../add-photos/add-photos';
+import { useStyles } from './styles';
+
+type NewItem = {
```
Comment (AleksandrTermenzhy) [inline packages/mobile/src/screens/new-item/components/new-item-form/new-item-form.tsx:18]: Don't we have a type yet in `shared`?

File: packages/mobile/src/screens/new-item/components/new-item-form/new-item-form.tsx [hunk 13]
```diff
@@ -0,0 +1,207 @@
...
+      </View>
+
+      <Text style={[globalStyles.fs14, globalStyles.mt6, styles.title]}>
+        {t('make_a_post.CONTACT')}
+      </Text>
+      <DropDown
+        label={t('make_a_post.COUNTRY')}
+        name="country"
+        control={control}
+        items={COUNTRIES}
```
Comment (AleksandrTermenzhy) [inline packages/mobile/src/screens/new-item/components/new-item-form/new-item-form.tsx:131]: For now it's only Ukraine, so we don't actually need dropdown here. Disabled input is seemed to be enough

File: packages/mobile/src/screens/new-item/components/new-item-form/styles.ts [hunk 14]
```diff
@@ -0,0 +1,52 @@
...
+const useStyles = () => {
+  const { dark, colors } = useCustomTheme();
+
+  return useMemo(
+    () =>
+      StyleSheet.create({
+        row: {
+          flexDirection: 'row',
+          width: '100%',
+          alignContent: 'flex-end',
```
Comment (AleksandrTermenzhy) [inline packages/mobile/src/screens/new-item/components/new-item-form/styles.ts:13]: You can use global styles instead

File: packages/mobile/src/components/components.ts (hunk 1 above)
Comment (seal323) [inline packages/mobile/src/components/components.ts:32]: move to top please

File: packages/mobile/src/localization/languages/ua/screens.ts (hunk 6 above)
Comment (natata7) [reply packages/mobile/src/localization/languages/ua/screens.ts:99]: we could have different phone codes for different countries

File: packages/mobile/src/common/types/types.ts [hunk 15]
```diff
@@ -20,3 +20,9 @@ export type {
 export type { ShowNotificationParams } from './notification/notification';
 export type { AppIcon, IconProps, CustomIconProps } from './ui/ui';
 export type { Theme, ThemeColors } from './theme/theme.type';
+export {
+  PopoverPlacement,
+  PopoverMode,
+  Rect as PopoverRect,
+  Size as PopoverSize,
+} from 'react-native-popover-view';
```
Comment (AleksandrTermenzhy) [inline packages/mobile/src/common/types/types.ts:28]: This should be at the top

File: packages/mobile/src/localization/languages/ua/screens.ts (hunk 6 above)
Comment (AleksandrTermenzhy) [inline packages/mobile/src/localization/languages/ua/screens.ts:99]: Again.. We don't need to translate this line)

File: packages/mobile/src/screens/new-item/components/new-item-form/new-item-form.tsx [hunk 16]
```diff
@@ -0,0 +1,220 @@
+import { ColorPalette, IPostForms } from '@vse-bude/shared';
```
Comment (AleksandrTermenzhy) [inline packages/mobile/src/screens/new-item/components/new-item-form/new-item-form.tsx:1]: This should go after the external imports

File: packages/mobile/src/screens/new-item/components/new-item-form/new-item-form.tsx [hunk 17]
```diff
@@ -0,0 +1,220 @@
...
+
+      <View
+        style={[
+          styles.row,
+          globalStyles.alignItemsCenter,
+          globalStyles.flexDirectionRow,
+        ]}
+      >
+        <Input
+          label=""
```
Comment (AleksandrTermenzhy) [inline packages/mobile/src/screens/new-item/components/new-item-form/new-item-form.tsx:105]: You allowed `undefined` here so:
```suggestion
```

File: packages/mobile/src/screens/new-item/components/new-item-form/new-item-form.tsx [hunk 18]
```diff
@@ -0,0 +1,220 @@
...
```
Comment (AleksandrTermenzhy) [inline packages/mobile/src/screens/new-item/components/new-item-form/new-item-form.tsx:183]:
```suggestion
```

File: packages/mobile/src/components/input/input.tsx [hunk 19]
```diff
@@ -19,17 +19,17 @@ import { globalStyles } from '~/styles/styles';
...
   errors: FormControlErrors<T>;
   placeholder?: string;
-  contentContainerStyle?: ViewStyle;
+  contentContainerStyle?: StyleProp<ViewStyle>;
   isSecure?: boolean;
 };
 
 const Input = <T extends FormControlValues>({
-  label,
+  label = '',
```
Comment (AleksandrTermenzhy) [inline packages/mobile/src/components/input/input.tsx:32]: This makes no sense as it will still be resolved to `false`

**Review by AleksandrTermenzhy — APPROVED** [review]
Resolve conflicts

**Pull request merged**